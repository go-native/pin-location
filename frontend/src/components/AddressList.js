import React from "react"
import AddressDetails from "./AddressDetails"

const AddressList = ({ addresses, onEdit, onRemove }) =>
  addresses.map(address => (
    <AddressDetails
      key={address.title}
      address={address}
      onEdit={onEdit}
      onRemove={onRemove}
    />
  ));

export default AddressList
