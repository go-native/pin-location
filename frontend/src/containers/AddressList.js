import React, { Component } from "react"
import { connect } from "react-redux"
import AddressList from "../components/AddressList"
import { fetchAddresses, openForm, removeAddresses } from "../actions"

class AddressListContainer extends Component {
  componentDidMount() {
    this.props.fetchAddresses();
  }
  render() {
    return <AddressList {...this.props} />;
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    addresses: state.addresses
  }
}
const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    fetchAddresses: () => dispatch(fetchAddresses()),
    onEdit: address => dispatch(openForm(address)),
    onEemove: id => dispatch(removeAddresses(id))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddressListContainer)
