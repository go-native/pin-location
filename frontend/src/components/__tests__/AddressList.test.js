import AddressList from "./../AddressList"
import React from "react"
import { shallow } from "enzyme"

const addresses = [{
  id: "1544109127653",
  title: "Leipziger str.",
  address: "Leipziger strasse",
  lat: 52.511123,
  lng: 13.399605
},{
  id: "1544109127654",
  title: "Spandauer str.",
  address: "Spandauer strasse",
  lat: 52.511144,
  lng: 13.399655
}]

describe("AddressList", () => {
  it("should render properly", () => {
    const wrapper = shallow(<AddressList addresses={addresses} />)
    expect(wrapper).toMatchSnapshot()
  })
})
