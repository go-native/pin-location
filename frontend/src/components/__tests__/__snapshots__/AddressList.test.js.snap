// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressList should render properly 1`] = `
Array [
  <AddressDetails
    address={
      Object {
        "address": "Leipziger strasse",
        "id": "1544109127653",
        "lat": 52.511123,
        "lng": 13.399605,
        "title": "Leipziger str.",
      }
    }
    key="Leipziger str."
  />,
  <AddressDetails
    address={
      Object {
        "address": "Spandauer strasse",
        "id": "1544109127654",
        "lat": 52.511144,
        "lng": 13.399655,
        "title": "Spandauer str.",
      }
    }
    key="Spandauer str."
  />,
]
`;
